vignettes/NSSP_nowcast.Rmd

```diff
@@ -0,0 +1,539 @@
+Filter to remove empty updates.
+```{r}
+nssp_merged <- merge(syn_nssp_time_stamps_long,
+  syn_nssp_diagnoses_long,
+  by = "event_id"
+) |>
+  filter(diagnoses_codes != ";;|") # drops empty updates
+```
+Now we have a dataframe where each row is an event, with the patient's visit start date (`C_Visit_date_Time`), the patient ID (`C_Processed_BioSense_ID`), the diagnoses code at the event (`diagnoses_code`), and the time stamp of the event (`time_stamp`).
+
+Next we will add a column for the time from arrival to each updated diagnosis, in days.
+```{r}
+nssp_updates <- nssp_merged |>
+  mutate(arrival_to_update_delay = as.numeric(difftime(
+    as.POSIXct(time_stamp), as.POSIXct(C_Visit_Date_Time),
+    units = "days"
+  )))
+```
+
+We next filter through the updates to find the first "hit" that corresponds to the diagnosis codes in the syndromic surveillance definition for ARI.
+```{r}
+ARI_updates <- nssp_updates |>
+  filter(map_lgl(diagnoses_codes, ~ any(str_detect(.x, diagnoses_codes_defn))))
+```
+
+Next, we will order these by the delay from visit to the diagnoses, and for each patient keep only the first update containing the ARI diagnoses code(s).
+```{r}
+first_ARI_diagnosis <- ARI_updates |>
+  arrange(arrival_to_update_delay) |>
+  group_by(C_Processed_BioSense_ID) |>
+  slice(1)
+```
+
+Label the visit start date, `C_Visit_Date_Time`, as the reference date,`reference_date` and `time_stamp` as the report date, `report_date` and remove the other column names that are no longer needed, as each row now represents a case.
+```{r}
+clean_line_list <- first_ARI_diagnosis |>
+  mutate(
+    reference_date = as.Date(C_Visit_Date_Time),
+    report_date = as.Date(time_stamp)
+  ) |>
+  ungroup()
+head(clean_line_list)
+```
+
+For nowcasting, we want to compute the number of incident cases indexed by reference and report date, so we can aggregate by reference and report date and compute the delay distribution.
+```{r}
+count_df_raw <- clean_line_list |>
+  group_by(reference_date, report_date) |>
+  summarise(count = n()) |>
+  mutate(delay = as.integer(report_date - reference_date))
+```
+Looking at this data, we can see that there is one case where there is a negative delay, which indicates that the time stamp of the diagnosis update was recorded before the start of the visit.
+This is likely due to a data entry error.
+We could choose to code these as having 0 delay, but if they are indeed erroneous this might bias our delay distribution estimation towards zero when they are equally likely to have had longer delays.
+We choose to exclude all negative valued delays.
+
+Filter out negative delays.
+```{r}
+count_df <- filter(count_df_raw, delay >= 0)
+head(count_df)
+```
+
+We have now generated data in the format that we need to use the `baselinenowcast` package, which requires a [long tidy dataframe](https://r4ds.had.co.nz/tidy-data.html) with incident case counts indexed by reference date and report date.
```

Comment:
link to something in epinowcast about the dataformat we nmeed and why maybe getting started etc.